packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -671,6 +671,14 @@ class _FocusState extends State<Focus> {
     Widget child = widget.child;
     if (widget.includeSemantics) {
       child = Semantics(
+        // Automatically request the focus for a focusable widget when it
+        // receives accessibility focus. Nothing is needed for losing the
+        // accessibility focus because if focus is lost, that means another node
+        // will gain focus and take focus from this widget.
+        onFocus:
+          !focusNode.hasFocus && focusNode.canRequestFocus
```

Comment:
Should this be `_couldRequestFocus ? focusNode.requestFocus : null`? It's strange that the condition that controls the availability of the `onFocus` callback is not exactly the same as the one used to control the `focusable` property below. If the two disagree, we could end up in a situation where the semantics thinks a widget is focusable while the framework does not, or vice versa.

If it is important to use separate conditions, it would also be good to have a test that covers it, and I may have to change my engine PR to do the right thing.